src/Configuration/Dotenv/File.hs

```diff
@@ -0,0 +1,38 @@
+-- |
+-- Module      :  Configuration.Dotenv.File
+-- Copyright   :  © 2015–2017 Stack Builders Inc.
+-- License     :  MIT
+--
+-- Maintainer  :  Stack Builders <hackage@stackbuilders.com>
+-- Stability   :  experimental
+-- Portability :  portable
+--
+-- This module provides parsers for dotenv files.
+
+module Configuration.Dotenv.File where
+
+import Control.Monad (liftM)
+import Control.Monad.IO.Class
+import System.Directory
+import Text.Megaparsec
+
+import Configuration.Dotenv.Parser
+import Configuration.Dotenv.Types
+
+-- | @parseMaybeFile@ parses a @.env@ file.
+-- If the file does not exist then it returns @Nothing@.
+parseMaybeFile :: MonadIO m => FilePath -> m (Maybe [Variable])
+parseMaybeFile path = liftIO $ do
+  exist <- doesFileExist path
```

Comment:
The `onMissingFile` approach is better because if it's not a subject to race conditions. If you test just before opening a file, you still may find that file missing a microsecond after. You should try to open it once and deal with exceptions, like it's done in `onMissingFile`. As I said I strongly recommend to put those functions you deleted back and preserve maximum backward-compatibility.